packages/commonjs/test/snapshots/function.js.md

```diff
@@ -3529,6 +3529,159 @@ Generated by [AVA](https://ava.li).
       `,
     }
 
+## named-imports-from-cjs
+
+> Snapshot 1
+
+    {
+      'main.js': `'use strict';␊
+      ␊
+      function unwrapExports (x) {␊
+      	return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;␊
+      }␊
+      ␊
+      function createCommonjsModule(fn, basedir, module) {␊
+      	return module = {␊
+      	  path: basedir,␊
+      	  exports: {},␊
+      	  require: function (path, base) {␊
+            return commonjsRequire(path, (base === undefined || base === null) ? module.path : base);␊
+          }␊
+      	}, fn(module, module.exports), module.exports;␊
+      }␊
+      ␊
+      function commonjsRequire () {␊
+      	throw new Error('Dynamic requires are not currently supported by @rollup/plugin-commonjs');␊
+      }␊
+      ␊
+      var foo = createCommonjsModule(function (module, exports) {␊
+      Object.defineProperty(exports, '__esModule', {␊
+        value: true␊
+      });␊
+      ␊
+      const a = 1;␊
+      const b = 2;␊
+      ␊
+      exports.a = a;␊
+      exports.b = b;␊
+      });␊
+      ␊
+      unwrapExports(foo);␊
+      var foo_1 = foo.a;␊
+      var foo_2 = foo.b;␊
+      ␊
+      const { a, b } = foo;␊
+      ␊
+      t.is(a, 1);␊
+      t.is(b, 2);␊
+      `,
+    }
+
+## named-imports-from-cjs-with-default
+
+> Snapshot 1
+
+    {
+      'main.js': `'use strict';␊
+      ␊
+      function unwrapExports (x) {␊
+      	return x && x.__esModule && Object.prototype.hasOwnProperty.call(x, 'default') ? x['default'] : x;␊
+      }␊
+      ␊
+      function createCommonjsModule(fn, basedir, module) {␊
+      	return module = {␊
+      	  path: basedir,␊
+      	  exports: {},␊
+      	  require: function (path, base) {␊
+            return commonjsRequire(path, (base === undefined || base === null) ? module.path : base);␊
+          }␊
+      	}, fn(module, module.exports), module.exports;␊
+      }␊
+      ␊
+      function commonjsRequire () {␊
+      	throw new Error('Dynamic requires are not currently supported by @rollup/plugin-commonjs');␊
+      }␊
+      ␊
+      var foo = createCommonjsModule(function (module, exports) {␊
+      Object.defineProperty(exports, '__esModule', {␊
+        value: true␊
+      });␊
+      ␊
+      const a = 1;␊
+      const b = 2;␊
+      ␊
+      exports.a = a;␊
+      exports.b = b;␊
+      exports.default = {a: 3, b: 4};␊
+      });␊
+      ␊
+      unwrapExports(foo);␊
+      var foo_1 = foo.a;␊
+      var foo_2 = foo.b;␊
+      ␊
+      const { a, b } = foo;␊
+      ␊
+      t.is(a, 1);␊
+      t.is(b, 2);␊
+      `,
+    }
+
+## named-imports-from-ejs
+
+> Snapshot 1
+
+    {
+      'main.js': `'use strict';␊
+      ␊
+      const a = 1;␊
+      const b = 2;␊
+      ␊
+      var foo = /*#__PURE__*/Object.freeze({␊
+      	__proto__: null,␊
+      	a: a,␊
+      	b: b␊
+      });␊
+      ␊
+      const { a: a$1, b: b$1 } = foo;␊
```

Comment:
I do not really understand. My expectation was that with this PR, the destructuring would be replaced by direct variable accesses, and the reified namespace object would not really be needed. But in all test cases, I still see the destructuring. And I fail to generate a test case where anything is improved. How come?

Reply:
Were you expecting `rollup` itself to optimize this case or the plugin?

Reply:
I was expecting the plugin create matching named exports and imports and replace the destructuring with those imports. Then Rollup would see no reason to create the namespace object.

At the moment, it is not really clear to me what this PR actually changes in terms of generated code. Could you give some examples with input code, generated output without this PR, generated output with this PR to get an idea of what this PR actually does?

Reply:
Ah, I think I understand now what I misunderstood.

Reply:
So what this really does is if there are named exports,

```
const { foo } = require('bar');
```

no longer becomes

```
import bar from 'bar';
const { foo } = bar;
```

but rather


```
import * as bar from 'bar';
const { foo } = bar;
```

Is that essentially correct?

Reply:
Interestingly I stumbled upon a bug in our logic here that predates your PR:

If I write

```js
exports.foo = 42;
exports.foo = 43;
```

in my imported file, Rollup throws a  "Duplicate export 'foo'" error. Alas...